modules/memfault-firmware-sdk/CMakeLists.txt

```diff
@@ -33,3 +33,7 @@ zephyr_library_sources_ifdef(
 zephyr_include_directories_ifdef(
         CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE
         config)
+
+zephyr_include_directories_ifdef(
+        CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE
+        config)
```

Comment:
The config folder gets added if you enable CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE, even though CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE is disabled. Can this cause side effects?

Reply:
No side effects! `memfault_platform_config_extra.h` will only get used if `CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE` is defined, which is a result of [this](https://github.com/memfault/memfault-firmware-sdk/blob/master/ports/zephyr/config/memfault_zephyr_platform_config.h#L82) check in the Zephyr platform config.